File: AmtOrk.Server/Models/Land.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.RegularExpressions;

namespace AmtOrk.Server.Models
{
	public class Land : RootModel
	{
		public string LandName { get; set; } = string.Empty;
		public string LandDescription { get; set; } = string.Empty;
		public string LandLocation { get; set; } = string.Empty;
		[ForeignKey("Kingdom")]
		public int KingdomId { get; set; }
		public string? LandIconPath { get; set; }
		public virtual AmtUser? Monarch { get; set; }
		public virtual AmtUser? Regent { get; set; }
		public virtual AmtUser? Champion { get; set; }
		public virtual AmtUser? PrimeMinister { get; set; }
		public virtual AmtUser? GMR { get; set; }

		public virtual List<Event>? LandEvents { get; set; }
		[NotMapped]
		public virtual string LandSlug { get => Regex.Replace(LandName.ToLower(), @"\s", "-"); }
	}
}

File: amtork.client/src/app/models/land.ts
import { AmtgardEvent } from "./amtgardEvent";
import { AmtUser } from "./amtUser";
import { Kingdom } from "./kingdom";

export interface Land
{
	id:number;
	landName:string;
	landDescription:string;
	landLocation:string;
	kingdomId:number;
	landIconPath:string | null;
	monarch:AmtUser | undefined;
	regent:AmtUser | undefined;
	champion:AmtUser | undefined;
	primeMinister:AmtUser | undefined;
	gmr:AmtUser | undefined;
	landEvents:AmtgardEvent[];
	landSlug:string;
}

export const emptyLand:Land = {
	id: 0,
	landName: "",
	landDescription: "",
	landLocation: "",
	kingdomId: 0,
	monarch: undefined,
	regent: undefined,
	champion: undefined,
	primeMinister: undefined,
	gmr: undefined,
	landEvents: [],
	landSlug: "",
	landIconPath: null
}